### PR TITLE
fix: Add wrapper around libbeat instrumentation to propagate missing tracer config

### PR DESCRIPTION
**Pull request by 1pkg**
## Motivation/summary

This PR addresses the [issue](https://github.com/elastic/apm-server/issues/13597) by adding a thin wrapper around libbeat instrumentation to propagate missing tracer config from elastic-agent via environmental variables.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [x] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

Follow instructions in `Building an Elastic Agent container image with a locally built APM Server`  and `Running an Elastic Agent container with a locally built APM Server` section in `TESTING.md`, update the relevant `agent.monitoring` sections in `elastic-agent.yml` to see the config are propagated down to tracing correctly.

## Related issues

#13597
#11381 
#13514

### CODE REVIEW

**Review by axw — DISMISSED** [review]
LGTM. Would be good (but perhaps not critical) to have a functional test for this. WDYT? Maybe open an issue to add a system test?

File: internal/beater/beater.go [hunk 1]
```diff
@@ -537,6 +537,50 @@ func (s *Runner) Run(ctx context.Context) error {
 	return errors.Join(result, closeErr)
 }
 
+// newInstrumentation is a thin wrapper around libbeat instrumentation that
+// sets missing tracer configuration from elastic-agent.
+func newInstrumentation(rawConfig *agentconfig.C) (instrumentation.Instrumentation, error) {
```
Comment (marclop) [inline internal/beater/beater.go:542]: A unit and/or system test would be great to have.

File: internal/beater/beater.go [hunk 2]
```diff
@@ -537,6 +537,50 @@ func (s *Runner) Run(ctx context.Context) error {
...
+	var apmCfg struct {
```
[the remark this answers is not in the document]
Comment (1pkg) [reply internal/beater/beater.go:550]: Added the comment + added the note about config keys re-marshaling from elastic-agent.

File: internal/beater/beater.go (hunk 1 above)
Comment (1pkg) [reply internal/beater/beater.go:542]: The test is added. Unfortunately, it's not possible to test this change with a unit test. So I added a small functional test instead that covers the added functionality.

**Review by marclop — APPROVED** [review]
Looks great! Thanks for adding the test.

File: changelogs/8.15.asciidoc [hunk 3]
```diff
@@ -35,3 +35,4 @@ https://github.com/elastic/apm-server/compare/v8.14.2\...v8.15.0[View commits]
 - Add require data stream to bulk index requests {pull}13398[13398]
 - Support self-instrumentation when in managed mode by getting tracing configs via reloader {pull}13514[13514]
 - Add mapping for OpenTelemetry attribute `messaging.destination.name` to derive `service.target` correctly {pull}13472[13472]
+- Propagate missing global_labels and tls tracing configs for self-instrumentation in managed mode {pull}13653[13653]
```
Comment (carsonip) [inline changelogs/8.15.asciidoc:38]: following @marclop 's suggestion
```suggestion
- Support self-instrumentation when in managed mode by getting tracing configs via reloader {pull}13514[13514] 13653[13653]
- Add mapping for OpenTelemetry attribute `messaging.destination.name` to derive `service.target` correctly {pull}13472[13472]
```

[the remark this answers is not in the document]
Comment (1pkg) [reply changelogs/8.15.asciidoc]: updated

Comment (1pkg) [reply changelogs/8.15.asciidoc]: updated

File: changelogs/8.15.asciidoc [hunk 4]
```diff
@@ -33,5 +33,5 @@ https://github.com/elastic/apm-server/compare/v8.14.2\...v8.15.0[View commits]
 - Upgraded bundled APM Java agent attacher CLI to version 1.50.0 {pull}13326[13326]
 - Enable Kibana curated UIs to work with hostmetrics from OpenTelemetry's https://pkg.go.dev/go.opentelemetry.io/collector/receiver/hostmetricsreceiver[hostmetricsreceiver] {pull}13196[13196]
 - Add require data stream to bulk index requests {pull}13398[13398]
-- Support self-instrumentation when in managed mode by getting tracing configs via reloader {pull}13514[13514]
-- Add mapping for OpenTelemetry attribute `messaging.destination.name` to derive `service.target` correctly {pull}13472[13472]
+- Support self-instrumentation when in managed mode by getting tracing configs via reloader {pull}13514[13514], {pull}13653[13653]
```
Comment (carsonip) [inline changelogs/8.15.asciidoc:36]: nit: we usually do it without `,`
```suggestion
- Support self-instrumentation when in managed mode by getting tracing configs via reloader {pull}13514[13514] {pull}13653[13653]
```

File: changelogs/8.15.asciidoc [hunk 5]
```diff
@@ -33,5 +33,5 @@ https://github.com/elastic/apm-server/compare/v8.14.2\...v8.15.0[View commits]
...
+- Add mapping for OpenTelemetry attribute `messaging.destination.name` to derive `service.target` correctly {pull}13472[13472]
```
Comment (carsonip) [inline changelogs/8.15.asciidoc:37]: nit: missing newline
```suggestion
- Add mapping for OpenTelemetry attribute `messaging.destination.name` to derive `service.target` correctly {pull}13472[13472]

```

File: internal/beater/beater.go [hunk 6]
```diff
@@ -537,6 +537,57 @@ func (s *Runner) Run(ctx context.Context) error {
 	return errors.Join(result, closeErr)
 }
 
+// newInstrumentation is a thin wrapper around libbeat instrumentation that
+// sets missing tracer configuration from elastic agent.
+func newInstrumentation(rawConfig *agentconfig.C) (instrumentation.Instrumentation, error) {
+	// This config struct contains missing fields from elastic agent APMConfig
+	// https://github.com/elastic/elastic-agent/blob/main/internal/pkg/core/monitoring/config/config.go#L127
+	// that are not directly handled by libbeat instrumentation bellow.
```
Comment (carsonip) [inline internal/beater/beater.go:545]:
```suggestion
	// that are not directly handled by libbeat instrumentation below.
```

**Review by carsonip — APPROVED** [review]
thanks for the unit test! left a few nits

**Review by carsonip — APPROVED** [review]

**Pull request merged**